zetaclient/chains/solana/observer/outbound.go

```diff
@@ -2,22 +2,283 @@ package observer
 
 import (
 	"context"
+	"fmt"
+	"math/big"
 
+	"cosmossdk.io/math"
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+	"github.com/pkg/errors"
 	"github.com/rs/zerolog"
 
-	"github.com/zeta-chain/zetacore/x/crosschain/types"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	"github.com/zeta-chain/zetacore/pkg/coin"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	"github.com/zeta-chain/zetacore/zetaclient/chains/interfaces"
+	zctx "github.com/zeta-chain/zetacore/zetaclient/context"
+	clienttypes "github.com/zeta-chain/zetacore/zetaclient/types"
+	"github.com/zeta-chain/zetacore/zetaclient/zetacore"
 )
 
 // GetTxID returns a unique id for Solana outbound
-func (ob *Observer) GetTxID(_ uint64) string {
-	return ""
+func (ob *Observer) GetTxID(nonce uint64) string {
+	tssAddr := ob.TSS().EVMAddress().String()
+	return fmt.Sprintf("%d-%s-%d", ob.Chain().ChainId, tssAddr, nonce)
```

Comment:
Can you elaborate on this?

Reply:
Let me move it to base observer and simplify it as `chainID-nonce`. It returns a unique identifier for outbound. For instance, the identifier will be used as key for maps.

Reply:
https://github.com/zeta-chain/node/pull/2560/commits/e94fd92e535981bea13dd89a2ad4e13a90a9c147